backend/src/main/java/ai/verta/modeldb/entities/versioning/DatasetRepositoryMappingEntity.java

```diff
@@ -0,0 +1,49 @@
+package ai.verta.modeldb.entities.versioning;
+
+import java.io.Serializable;
+import java.util.Objects;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "dataset_repository_mapping")
+public class DatasetRepositoryMappingEntity implements Serializable {
+  public DatasetRepositoryMappingEntity() {}
+
+  public DatasetRepositoryMappingEntity(
+      RepositoryEntity repositoryEntity, boolean isDatasetRepository) {
+    this.repositoryEntity = repositoryEntity;
+    this.isDatasetRepository = isDatasetRepository;
+  }
+
+  @Id
+  @OneToOne(cascade = CascadeType.ALL)
+  @JoinColumn(name = "repository_id")
+  private RepositoryEntity repositoryEntity;
+
+  @Column(name = "is_dataset_repository")
+  private Boolean isDatasetRepository = false;
```

Comment:
is the second column required?
we caqn only store the repos which are dataset , why store rows which have this as false?

Reply:
Fixed